instrumentation/play-2.6.13/src/main/scala/play/core/server/Server_Instrumentation.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ *
+ *  * Copyright 2020 New Relic Corporation. All rights reserved.
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package play.core.server
+
+import com.newrelic.api.agent.weaver.Weaver
+import com.newrelic.api.agent.weaver.scala.{ScalaMatchType, ScalaWeave}
+import com.newrelic.api.agent.{NewRelic, Trace}
+import com.nr.agent.instrumentation.play26.PlayUtil
+import play.api.mvc.{Handler, RequestHeader, Result}
+import play.core.ApplicationProvider
+
+import scala.concurrent._;
+
+@ScalaWeave(`type` = ScalaMatchType.Object, `originalName` = "play.core.server.Server")
+class Server_Instrumentation {
+  @Trace(dispatcher = true)
+  def getHandlerFor(
+    request: RequestHeader,
+    applicationProvider: ApplicationProvider
+  ): Either[Future[Result], (RequestHeader, Handler)] = {
+    val tempResult: Either[Future[Result], (RequestHeader, Handler)] = Weaver.callOriginal()
+    val result = tempResult.right.map { case (request, handler) =>
```

Comment:
I'm not very familiar with Scala or Play. Is this assignment of result always modifying the original and is that safe?

Reply:
Most of the default operations and data structures in scala are immutable.  This operation will create a new object entirely and will not modify tempResult at all.